### DISCUSSION TITLE
Development is kicking: Road to v2 🚀

### DISCUSSION THREAD

**Discussion by andygrunwald** [andygrunwald/go-jira#501] [General]
Discussion around https://github.com/andygrunwald/go-jira/issues/489

Let us know what you think!

**Comment by catouc** [comment]
@andygrunwald I just had a thought while developing a CLI against Jira with a much smaller scope than https://github.com/go-jira/jira but it would be nice to have the types isolated into their own package so they can be used without pulling in the entire client :)

[Flux](https://github.com/fluxcd/pkg/tree/main/apis) is doing that quite nicely and it makes developing against established type definitions a lot easier without having to copy paste the code with the right attributions.

Would also help out with the code if time is an issue - just let me know!

**Comment by benjivesterby** [comment]
@andygrunwald I'm starting up work on my integration with JIRA again and was going to jump in on some v2 stuff. Where is it currently? I don't see a branch in the go-jira repo.

**Reply by andygrunwald** [reply]
@benjivesterby The `main` branch is already the WIP v2 branch, see https://github.com/andygrunwald/go-jira#warning-state-of-this-library-warning - PRs against the `main` branch incl. a changelog / migration entry should be ok.

**Comment by benjivesterby** [comment]
@andygrunwald for v2 we should consider using this: https://github.com/devnw/ajson

It's a derivative of your initial implementation for dealing with unknown JSON fields on issues but it supports recursive values and generic types. It also uses the new experimental json library that the Go team has been working on.

**Reply by andygrunwald** [reply]
@benjivesterby I don't know the library itself, but a quick look might be a useful thing, especially for the custom fields (which is an often mentioned pain).
I am not sure if this should be used for standard fields (e.g., for an Issue). Mainly because those should be explicit.

**Comment by YannickAlex07** [comment]
Hey @andygrunwald, long time no see! Could you provide any update on the current state of this? We are looking into using this package for some of our projects, however it looks like the development is kinda stale right now, no? Any info is appreciated!

**Reply by andygrunwald** [reply]
@YannickAlex07 INdeed. Hope you are doing well.
Am I planning to work on this library at some point and continue to move it to a good usable state? Yes. At the moment my time is limited due to other obligations. There won't be much movement in the next month.

If you are planning to use this library in your company, you can do it right now. Many basic functionality is given and usable.
If you need additional functionally, feel free to add a PR. It might take a bit until I am able to properly review it.
If you need additional functionality quick, we can talk about an agreement if needed.

Does this answer your question?

**Reply by YannickAlex07** [reply]
Yes this answers my question perfectly! We will try it out as soon as we get to it - tbh the project were I want to use this is moving slowly anyway, so there is no pressure from my side here. Thanks for the reply!
